torch/testing/_internal/common_modules.py

```diff
@@ -211,6 +211,7 @@ def __init__(self,
                  skips=(),  # Indicates which tests to skip
                  decorators=None,  # Additional decorators to apply to generated tests
                  dtypes=floating_types(),  # dtypes this function is expected to work with
+                 dtypesIfMPS=(torch.float16, torch.float32,),  # dtypes this function is expected to work with on MPS
```

Comment:
I am curious how we would scale this in the future when there may be more accelerators. I'd guess we'd have an API somewhat like `accelerator.get_supported_dtypes()` and the `dtypes` field here could be a dynamic function that gets the current accelerator and returns the relevant dtypes, similar to your supported_dtypes function.

This then brings me to think "hey, maybe in the future there would be an accelerator-aware `supported_floating_types()` vs what's used in line 213". Anyway, this discussion is likely outside the scope of this PR. cc'ing @albanD in case he has thoughts

Reply:
Great point! I was thinking about this too :)

I don't think we will have BC issues changing the dtype logic in the test infra. Updating `dtypes` to account for the current accelerator in the future definitely sounds reasonable to me